File: ruby/word-count/word_count.rb
class Phrase
  def initialize(phrase)
    @phrase = phrase
  end

  def word_count
    words = Hash.new(0)
    @phrase.split(/[,\.\s]+/).each do |word|
      word = word.sub(/^\W+/, "").sub(/\W+$/, "").downcase
      next if word == ''
      words[word] += 1
    end
    return words
  end
end

File: python/word-count/word_count.py
import re
def count_words(sentence):
    words = re.split(r"[\s:!\"_,]+", sentence)
    resp = {}
    for word in words:
        word = word.lower()
        word = re.sub(r"(^'+|'+$|[.,&@$%^]+)", "", word)
        if word != '':
            resp[word] = 1 if word not in resp else resp[word] + 1
    return resp

